Add Logicmonitor exporter overall structure

khyatigandhi6 opened the pull request:
**Description:** <Describe what has changed.>
Adding new component, Logicmonitor exporter.

This is the first of 3 PRs which adds the overall skeleton structure.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/13727

**Testing:** Unit Tests

**Documentation:**  Readme explaining the exporter and its configuration.

khyatigandhi6 committed:
Add Logicmonitor Exporter overall structure

khyatigandhi6 committed:
Initiate default retry and queue settings

khyatigandhi6 committed:
Add Status details in README.md

bogdandrutu commented on exporter/logicmonitorexporter/config.go:
```diff
@@ -0,0 +1,67 @@
+	// ApiToken of LM Platform
+	APIToken map[string]string `mapstructure:"apitoken"`
+
+	// Field to enable/disable batching for logs
+	LogBatchingEnabled bool `mapstructure:"log_batching_enabled"`
+
+	// Batching interval for logs
+	LogBatchingInterval time.Duration `mapstructure:"log_batching_interval"`
```
We recommend to not duplicate this logic, but recommend the use of the batch processor.

khyatigandhi6 replied:
Removed the batching logic.

bogdandrutu commented on exporter/logicmonitorexporter/testdata/config.yaml:
```diff
@@ -0,0 +1,23 @@
+receivers:
```
We do not recommend using full config for testing, see an example here https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/opencensusreceiver/testdata/config.yaml and you can check the usage https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/opencensusreceiver/config_test.go#L32

khyatigandhi6 replied:
Updated the PR taking reference from the above suggested example.

khyatigandhi6 committed:
Remove batching and replace full config
Removed batching logic from logs exporter.
Replaced full test config.yaml with component specific details.

bogdandrutu commented on exporter/logicmonitorexporter/config.go:
```diff
@@ -0,0 +1,52 @@
+	exporterhelper.RetrySettings `mapstructure:"retry_on_failure"`
+	ResourceToTelemetrySettings  resourcetotelemetry.Settings `mapstructure:"resource_to_telemetry_conversion"`
+
+	// ApiToken of LM Platform
+	APIToken map[string]string `mapstructure:"apitoken"`
```
are only `access_id` and `access_key` possible values? If yes, create a struct with these 2 values instead of `map[string]string`.

khyatigandhi6 replied:
Yes these are only possible values. Addressed the requested changes.

khyatigandhi6 committed:
Update dependencies to v0.66.0 and review comment change in config.go

bogdandrutu commented:
https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3571261425/jobs/6003205884

khyatigandhi6 committed:
Fix failed check of ISSUE TEMPLATE

khyatigandhi6 committed:
Upgrade collector version and fix deprecated functions
